### PR TITLE
[wip] fix(storage-proofs-porep): switch to hwloc2

### PR DESCRIPTION
**Pull request by dignifiedquire**

### CODE REVIEW

File: storage-proofs/porep/src/stacked/vanilla/cores.rs [hunk 1]
```diff
@@ -62,44 +62,50 @@ fn get_thread_id() -> ThreadId {
...
+    prior_state: Option<hwloc2::Bitmap>,
 }
 
 impl Drop for Cleanup {
     fn drop(&mut self) {
         if let Some(prior) = self.prior_state.take() {
-            let child_topo = &TOPOLOGY;
-            let mut locked_topo = child_topo.lock().expect("poisded lock");
-            let _ = locked_topo.set_cpubind_for_thread(self.tid, prior, CPUBIND_THREAD);
+            if let Some(ref mut locked_topo) = &mut *TOPOLOGY.lock().expect("poisded lock") {
```
Comment (porcuquine) [inline storage-proofs/porep/src/stacked/vanilla/cores.rs:71]: Let's fix that typo while we're making changes here.

**Review by porcuquine — APPROVED** [review]

File: storage-proofs/porep/src/stacked/vanilla/cores.rs (hunk 1 above)
Comment (porcuquine) [reply storage-proofs/porep/src/stacked/vanilla/cores.rs:71]: I approved on the assumption this is intended to just make mechanical changes to facilitate library version upgrade. CI is failing, though, so further tweaking may be required.

Comment (dignifiedquire) [reply storage-proofs/porep/src/stacked/vanilla/cores.rs:71]: all CI is failing atm, will work on that after that is fixed in #1358

**Comment by ozhtdong** [comment]
It's strange that the hwloc2 identifies NUMA nodes as cores. And all object types are messed up in all depthes.

**Comment by vmx** [comment]
This PR hasn't been touched in years, hence closing it.

**Pull request closed without merging**